corehq/ex-submodules/casexml/apps/phone/data_providers/standard.py

```diff
@@ -36,15 +36,30 @@ def get_elements(self, restore_state):
         yield xml.get_registration_element(restore_state.user)
 
 
-class FixtureElementProvider(RestoreDataProvider):
+class FixtureElementMixin(object):
     """
     Gets any associated fixtures.
     """
+    def __init__(self, fixture_group):
+        self.fixture_group = fixture_group
+
     def get_elements(self, restore_state):
         # fixture block
         for fixture in generator.get_fixtures(
             restore_state.user,
             restore_state.version,
-            restore_state.last_sync_log
+            restore_state.last_sync_log,
+            group=self.fixture_group
         ):
             yield fixture
+
+
+class FixtureElementProvider(FixtureElementMixin, RestoreDataProvider):
+    pass
+
+
+class FixtureResponseProvider(FixtureElementMixin, LongRunningRestoreDataProvider):
+    def get_response(self, restore_state):
+        response = restore_state.restore_class()
+        response.extend(self.get_elements(restore_state))
+        return response
```

Comment:
Awesome!

Let me know when this is up on staging so we can start running tests?

-Clayton

On Wed, Jun 3, 2015 at 11:20 AM, dimagimon <notifications@github.com> wrote:

> In corehq/ex-submodules/casexml/apps/phone/data_providers/standard.py
> <https://github.com/dimagi/commcare-hq/pull/6952#discussion_r31634522>:
>
> >          ):
> >              yield fixture
> > +
> > +
> > +class FixtureElementProvider(FixtureElementMixin, RestoreDataProvider):
> > +    pass
> > +
> > +
> > +class FixtureResponseProvider(FixtureElementMixin, LongRunningRestoreDataProvider):
> > +    def get_response(self, restore_state):
> > +        response = restore_state.restore_class()
> > +        response.extend(self.get_elements(restore_state))
> > +        return response
>
> W292 no newline at end of file
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/6952/files#r31634522>.
>